### DISCUSSION TITLE
one-to-one relationship shown as one-to-many in ERD

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [dbeaver/dbeaver#21389] [Q&A]
I have a SQLite database.

Table A has primary key `A.id`.

Table B has column `B.A_id`, which is a foreign key referencing `A.id`. `B.A_id` is declared as `FOREIGN KEY(A_id) REFERENCES A(id)`. It is also constrained to be `UNIQUE`.

Thus there is a one-to-one relationship between A and B.

However, the ERD shows it as one-to-many, with the "many" side on  `B.A_id`.
![image](https://github.com/dbeaver/dbeaver/assets/54762742/68b9a678-8744-444f-bd38-6361630fe4e1)

Why is that? How is a one-to-one relationship recognized?

EDIT:

I have double-checked with:
```
SELECT MAX(count) FROM (
SELECT COUNT(*) AS count FROM B
GROUP BY A_id);
```

That `A_id` is unique.

**Comment by mkgrgis** [comment]
ISO SQL treat  1↔1 as foreign key to group of columns equal to primary key group of columns in source table. Look like ERD really need improvement.

**Comment by E1izabeth** [comment]
Thanks for reporting
https://github.com/dbeaver/dbeaver/issues/21398
